src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -4891,6 +4906,416 @@ void TestAction(IFormFile? file, TraceIdentifier traceId)
         Assert.Equal("my-trace-id", traceIdArgument.Id);
     }
 
+    public static TheoryData<HttpContent, string> FormContent
+    {
+        get
+        {
+            var dataset = new TheoryData<HttpContent, string>();
+
+            var multipartFormData = new MultipartFormDataContent("some-boundary");
+            multipartFormData.Add(new StringContent("hello"), "message");
+            multipartFormData.Add(new StringContent("foo"), "name");
+            dataset.Add(multipartFormData, "multipart/form-data;boundary=some-boundary");
+
+            var urlEncondedForm = new FormUrlEncodedContent(new Dictionary<string, string> { ["message"] = "hello", ["name"] = "foo" });
+            dataset.Add(urlEncondedForm, "application/x-www-form-urlencoded");
+
+            return dataset;
+        }
+    }
+
+    [Fact]
+    public void CreateThrowsNotSupportedExceptionIfIFormCollectionHasMetadataParameterName()
+    {
+        IFormCollection? formArgument = null;
+
+        void TestAction([FromForm(Name = "foo")] IFormCollection formCollection)
+        {
+            formArgument = formCollection;
+        }
+
+        var nse = Assert.Throws<NotSupportedException>(() => RequestDelegateFactory.Create(TestAction));
+        Assert.Equal("Assigning a value to the IFromFormMetadata.Name property is not supported for parameters of type IFormCollection.", nse.Message);
+    }
+
+    [Theory]
+    [MemberData(nameof(FormContent))]
+    public async Task RequestDelegatePopulatesFromIFormCollectionParameter(HttpContent content, string contentType)
+    {
+        IFormCollection? formArgument = null;
+
+        void TestAction(IFormCollection formCollection)
+        {
+            formArgument = formCollection;
+        }
+
+        var stream = new MemoryStream();
+        await content.CopyToAsync(stream);
+
+        stream.Seek(0, SeekOrigin.Begin);
+
+        var httpContext = CreateHttpContext();
+        httpContext.Request.Body = stream;
+        httpContext.Request.Headers["Content-Type"] = contentType;
+        httpContext.Features.Set<IHttpRequestBodyDetectionFeature>(new RequestBodyDetectionFeature(true));
+
+        var factoryResult = RequestDelegateFactory.Create(TestAction);
```

Comment:
Let's add at least one test to ensure the metadata gets added if you call `RequestDelegateFactory.InferMetadata`.